lib/vdsm/storage/sdm/api/merge.py

```diff
@@ -42,6 +42,14 @@ def _run(self):
             with self.subchain.prepare(), self.subchain.volume_operation():
                 top_vol_path = self.subchain.top_vol.getVolumePath()
                 base_vol_path = self.subchain.base_vol.getVolumePath()
+
+                # There may be stale bitmaps in base if they were left
+                # after failed delete bitmap operation. These bitfields may
+                # cause the bitfields merge operation to fail with ENOSPC.
+                # As there is not a reliable way to calculate the size of
+                # these bitfields, we prune them before the commit.
+                bitmaps.prune_bitmaps(base_vol_path, top_vol_path)
```

Comment:
It could be better to prune bitmaps when we prepare for merge, since we measure and extend base volume at that point. But we know that measure does not consider the bitmaps in base, so pruning before or after measure is the same. If qemu-img measure will start to consider bitmaps in base, we may over allocate the base volume before merge. If we want to avoid this issue, we can move pruning to the prepare step (vdsm.storage.merge.prepare).